src/scripting_api/code_gen.js

```diff
@@ -0,0 +1,129 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Event, EventDispatcher } from "./event.js";
+import { AForm } from "./aform.js";
+import { App } from "./app.js";
+import { Console } from "./console.js";
+import { Doc } from "./doc.js";
+import { Field } from "./field.js";
+import { NotSupportedError } from "./error.js";
+import { PDFObject } from "./pdf_object.js";
+import { ProxyHandler } from "./proxy.js";
+import { Util } from "./util.js";
+
+function generateRandomString(actions) {
+  while (true) {
+    const name = new Uint8Array(64);
+    if (typeof crypto !== "undefined") {
+      crypto.getRandomValues(name);
+    } else {
+      for (let i = 0, ii = name.length; i < ii; i++) {
+        name[i] = Math.floor(256 * Math.random());
+      }
+    }
+
+    const nameString = btoa(
+      Array.from(name)
+        .map(x => String.fromCharCode(x))
+        .join("")
+    );
+    if (actions.every(action => !action.includes(nameString))) {
+      return nameString;
+    }
+  }
+}
+
+function dumpClass(clazz) {
+  return clazz
+    .toString()
+    .replace(/extends _[^\.]*\./, "extends ")
+    .replace(/new _[^\.]*\./, "new ")
+    .replace(/throw new _error\./g, "throw new ");
+}
+
+function generateCode({ document, objects, calculationOrder }) {
```

Comment:
As mentioned above, is there anything keeping us from just loading all the script as text and evaling in the sandbox?